### PR TITLE
ref(platformIntegrationSetup): Convert class to func

### PR DESCRIPTION
**Pull request by priscilawebdev**

### CODE REVIEW

File: static/app/views/projectInstall/platformIntegrationSetup.tsx [hunk 1]
```diff
@@ -1,172 +1,137 @@
...
+  const [installed, setInstalled] = useState(false);
+  const navigate = useNavigate();
+
+  const {data: integrations} = useApiQuery<{providers: IntegrationProvider[]}>(
+    [
+      `/organizations/${organization.slug}/config/integrations/?provider_key=${integrationSlug}`,
+    ],
+    {
+      enabled: !!integrationSlug,
+      staleTime: Infinity,
```
Comment (priscilawebdev) [inline static/app/views/projectInstall/platformIntegrationSetup.tsx:45]:
```suggestion
      staleTime: 0,
```

File: static/app/views/projectInstall/platformIntegrationSetup.tsx [hunk 2]
```diff
@@ -1,172 +1,137 @@
...
+    window.scrollTo(0, 0);
+    // redirect if platform is not known.
+    if (!platform || platform.id === 'other') {
+      redirectToNeutralDocs();
     }
+  }, [redirectToNeutralDocs, platform]);
+
+  if (!provider || !platform || !project) {
+    return null;
+  }
```
Comment (scttcper) [inline static/app/views/projectInstall/platformIntegrationSetup.tsx:71]: I think there needs to be a loading state, the AsyncComponent before was handling this

Comment (priscilawebdev) [reply static/app/views/projectInstall/platformIntegrationSetup.tsx:71]: oh yes sure! always when I convert a component that was previously using `AsyncComponent` I forget about the loading state. Thanks 🙏

File: static/app/views/projectInstall/platformIntegrationSetup.tsx [hunk 3]
```diff
@@ -1,172 +1,150 @@
...
+      `/organizations/${organization.slug}/config/integrations/?provider_key=${integrationSlug}`,
+    ],
+    {
+      enabled: !!integrationSlug,
+      staleTime: 0,
+    }
+  );
 
+  useEffect(() => {
+    window.scrollTo(0, 0);
```
Comment (ArthurKnaus) [inline static/app/views/projectInstall/platformIntegrationSetup.tsx:59]: I would move the `scrollTo` into a separate `useEffect` to make sure it only runs on mount.

**Review by ArthurKnaus — APPROVED** [review]

**Pull request merged**